### PR TITLE
SVG image support

### PR DESCRIPTION
**Pull request by jams2**
Implementation of SVG support for [RFC 77](https://github.com/wagtail/rfcs/blob/svgs/text/077-svg.md). There was a fair amount of [discussion](https://github.com/wagtail/rfcs/pull/77), and the RFC was not subsequently updated, so I've done my best to capture the feedback. I will outline where the PR deviates from the initial spec below.

This feature requires my [Willow SVG branch](https://github.com/wagtail/Willow/pull/99). That PR has discussion on the security issues related to parsing SVGs (and XML in general), and on the choice to not include a rasterisation plugin initially (in short - we can crop and resize SVGs without rasterising them).

I've tested that SVGs behave as expected when rendered in templates via the `image` template tag (with all of the formatting options), when included in rich text, and when accessed via the Wagtail v2 API.

Attached is a PDF demonstrating the rendering of SVGs alongside raster images in various formats (the French Alps photo is a raster image, the others are SVGs): [svg-test.pdf](https://github.com/wagtail/wagtail/files/10338795/svg-test.pdf).

## SVGs in templates

I changed the `image` template tag so that it takes an optional `preserve-svg` argument. If this argument is passed, and the image is an SVG, operations that would require the SVG to be rasterised are not applied (as this will cause an error without a rasterisation library).

The RFC originally proposed to add a `source` argument to the `image` template tag. Feedback was that this might be confusing for developers (as `img` elements have a `src` attribute). Additionally, I'm not sure what value this adds when the "original" rendition is available.

I considered additionally adding a `WAGTAILIMAGES_PRESERVE_SVGS` setting to unconditionally preserve SVGs when generating renditions, but due to the fact that renditions are generated in a few different code paths, it feels that this would be a drag to maintain.

## SVGs in rich text

The default rich text image format does not require rasterisation of SVGs for generating renditions, so an equivalent to `preserve-svg` is not required - SVGs will work out of the box. I feel that developers implementing custom rich text image formats can handle this if necessary, using the provided `to_svg_safe_spec` utility.

## SVGs in the API

The `ImageRenditionField` API field allows developers to specify a filter spec for the field, so I added a `preserve_svg` option that makes the rendition generation process behave equivalently to the `image` template tag with `preserve-svg`.

## Issues

SVGs without explicit width/height attributes do not preview on the image upload page. I have not had time to look into this, and will be AFK from Jan 4 - Jan 27.

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: docs/topics/images.md
Comment (jacobtoppm) [inline docs/topics/images.md:428]: Can you add this to the settings reference too please?

**Review by jacobtoppm — COMMENTED** [review]
Some great work from a look so far

File: docs/topics/images.md
Comment (jams2) [reply docs/topics/images.md]: I've removed this subsequent to #9860 - see https://github.com/wagtail/wagtail/pull/9842/commits/5812c0763d50320924ef239fdc6b19413f49d8b3

File: docs/topics/images.md
Comment (jams2) [reply docs/topics/images.md]: I've added some specifics in https://github.com/wagtail/wagtail/pull/9842/commits/f90388ed395710a31594724857c94a29f63d629f.

The fact that a user will not be protected from a malicious SVG if they download the file and then subsequently open it in the browser makes me a little nervous, but:
- allowing SVGs is opt-in;
- the docs are clear about the risks;
- the execution environment in this scenario is not the Wagtail site's domain; and
- access to file system and media device APIs requires user interaction/approval.

I'm curious to hear others' thoughts on this.

I'd also like to add support for sanitising SVGs, probably as a Willow plugin.

**Comment by jams2** [comment]
Above issue on Willow PR has been resolved.

**Review by gasman — APPROVED** [review]
Looks good to me! Will merge once the corresponding updates to Willow are in.

File: wagtail/images/templatetags/wagtailimages_tags.py
Comment (gasman) [inline wagtail/images/templatetags/wagtailimages_tags.py:2]: Looks like `functools.cache` was [only added in Python 3.9](https://docs.python.org/3/library/functools.html#functools.cache), so we can't use it in Wagtail until 3.8 is EOL.

**Review by gasman — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
